cli/src/main/kotlin/org/jetbrains/intellij/pluginRepository/client.kt

```
@@ -50,10 +50,14 @@ class Client {
             Args.parseOrExit(options, args)
             val pluginRepository = PluginRepositoryInstance(options.host, options.username!!, options.password!!)
             val pluginId = options.pluginId!!
-            if (pluginId.matches(Regex("\\d+"))) {
-                pluginRepository.uploadPlugin(pluginId.toInt(), File(options.pluginPath!!), parseChannel(options.channel))
-            } else {
-                pluginRepository.uploadPlugin(pluginId, File(options.pluginPath!!), parseChannel(options.channel))
+            try {
+                if (pluginId.matches(Regex("\\d+"))) {
+                    pluginRepository.uploadPlugin(pluginId.toInt(), File(options.pluginPath!!), parseChannel(options.channel))
+                } else {
+                    pluginRepository.uploadPlugin(pluginId, File(options.pluginPath!!), parseChannel(options.channel))
+                }
+            } catch (e: Exception) {
+                System.exit(1)
```

Any logging here? And probably you want to catch Throwable, not Exception

---

`uploadPlugin` contains logging. 
Not sure about Throwable. Exception here is an expected situation (host not reachable, file not found, e.t.c) and Error is absolutely unexpected, so stacktrace output may be useful.

---

I mean the code could change at later point, you may not handled all situations, etc etc
So generally you do not ignore exceptions. If you handle them elsewhere, drop catch here,
if you're not sure, insert logging

---

I don't understand what kind of logging do you expect here. We are calling a single function and this function already logs all useful information. 
If I'll drop catch here then user will see the full stacktrace of exception and I don't think that this is a good output, because exception can be thrown in a very simple cases, for example, when file doesn't exist.

---

@satamas You ignore 'e' here, so if `uploadPlugin` throws an exception and doesn't log it the program will exit without printing any error message.

---

@satamas I think the problem is in `uploadPluginInternal` functions: it should either don't log the exception or don't rethrow it.

---

This function is used in a different places (at least in gradle plugin and in cli). From my point of view this means that it must throw an exception on failed upload to indicate that something is wrong, otherwise it would be easy to ignore failed upload.
On the other hand we have logging and we are writing to log that we've started plugin uploading. It would be strange if wouldn't write the result of plugin uploading then.
I've narrowed suppressed exceptions to `RetrofitError`, any other exceptions would be printed to console. I think that this is a good compromise.

---

@satamas Such code creates non-evident requirements for `uploadPlugin` functions: them must log all exception thrown from they. It's better to avoid such implicit requirements because it's easy to violate them. Note that actually we already have a problem here: exceptions thrown in `ensureCredentialsAreSet` are already silently ignored.

---

@chashnikov as @satamas stated suppressed exceptions are narrowed to `RetrofitError` because non 200 responses from server produce this exception and it is a normal requirement for `uploadPlugin` to process non 200 responses. `ensureCredentialsAreSet` throws `RuntimeException` which is not ignored.